iree/compiler/Dialect/HAL/Target/WebGPU/SPIRVToWGSL.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/HAL/Target/WebGPU/SPIRVToWGSL.h"
+
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+#include "tint/tint.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace HAL {
+
+llvm::Optional<std::string> compileSPIRVToWGSL(
+    llvm::ArrayRef<uint32_t> spvBinary) {
+  // TODO(scotttodd): reroute to MLIR diagnostics?
+  auto diagPrinter = tint::diag::Printer::create(stderr, true);
+  tint::diag::Formatter diagFormatter;
+
+  // TODO(scotttodd): remove this copy (API for std::span or [uint8_t*, size]?)
```

Comment:
would be good ergonomics feedback to them - there's no reason they need an std::vector and an overload that took a pointer + size would be nice until std::span (they take it const& so they are either making a copy or don't need it to survive after the call)